### ISSUE TITLE
Logical image() argument names & defaults to make images look like images

### ISSUE THREAD

**Issue by ianshmean** [JuliaPlots/Makie.jl#524]
Related to https://github.com/JuliaPlots/AbstractPlotting.jl/pull/98, https://github.com/JuliaPlots/Makie.jl/issues/295 and  https://github.com/JuliaPlots/Makie.jl/issues/205

With the caveat of not being someone with familiarity of the inner workings.. here's my suggestion for `image()` functionality, to make images appear like I think most people would expect them to. I've aimed for logical keywords that I might try _before looking at the docs_...

Defaults for `image()`:
- `transpose::Bool=true` - Swaps the x & y axes. Happens before the options below
- `flipy::Bool=true` - Prevents images looking upside down
- `flipx::Bool=false` - A good option to have access to. i.e. front-facing cameras where `flipx` is common practice
- `squarepixels::Bool=true` - The equivalent of `scale_plot = false`.
- `show_axis::Bool=false` - I think most would expect image() to show without the axes & grid

@ssfrr @SimonDanisch

**Comment by KronosTheLate** [comment]
I agree that it would feel right if `image` showed something that was flipped and tansposed (i.e. `rotr90`) compared to say a heatmat. I though that would be half the point of having both `image` and `heatmap`.

Are there anyone thinking that it a bad idea?

**Comment by SimonDanisch** [comment]
Nope, I think everyone is on board with that, I just need to find some time to implement it...

**Comment by ffreyer** [comment]
#1406 implemented some of the transformations in a roundabout way.
